src/overlay/src/Overlay.js

```diff
@@ -90,6 +90,14 @@ class Overlay extends React.Component {
      */
     shouldCloseOnEscapePress: PropTypes.bool,
 
+    /**
+     * Function called when overlay is about to close.
+     * Return `true` to close the sheet. (default)
+     * Return `false` to prevent the sheet from closing.
+     * type: `Function -> Boolean`
+     */
+    onBeforeClose: PropTypes.func,
```

Comment:
Would it make more sense to call this `onRequestClose`? Where only explicit `false` returns would prevent closing?

Reply:
Hey Matt!  I know you have some similar patterns like this in `Pagination` currently, and I think there is a real use case for it. But for this I think `onBeforeClose` sounds slightly simpler and more in line with browser events such as `onbeforeunload`.